_pytest/unittest.py

```diff
@@ -157,9 +156,20 @@ def runtest(self):
             self._testcase(result=self)
         else:
             # disables tearDown and cleanups for post mortem debugging (see #1890)
+            # but still implements the skipping machinery (see #2137)
+            testMethod = getattr(self._testcase, self._testcase._testMethodName)
+            if (getattr(self._testcase.__class__, "__unittest_skip__", False) or
+                getattr(testMethod, "__unittest_skip__", False)):
+                # If the class or method was skipped.
+                skip_why = (getattr(self._testcase.__class__, '__unittest_skip_why__', '') or
+                            getattr(testMethod, '__unittest_skip_why__', ''))
+                try:
```

Comment:
I recommend making an explicit version check here; try/except (specially on a generic exception like `TypeError`) can catch an exception on some unintended part of the code, making it hard to understand/debug the problem.

In short:

```python
if sys.version_info[0] == 3:
    self._testcase._addSkip(self, self._testcase, skip_why)
else:
    self._testcase._addSkip(self, skip_why)
```